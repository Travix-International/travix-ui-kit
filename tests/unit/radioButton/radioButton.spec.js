import * as enzyme from 'enzyme';
import * as enzymeToJson from 'enzyme-to-json';
import React from 'react';
import RadioButton from '../../../components/radioButton/radioButton';

describe('Radio Button', () => {
  describe('#render()', () => {
    const shallow = enzyme.shallow;
    const shallowToJson = enzymeToJson.shallowToJson;
    const onChange = () => {};

    it('should render disabled radio button', () => {
      const wrapper = shallow(
        <RadioButton disabled id="disabledRadio" onChange={onChange}>
          Disabled radio button
        </RadioButton>
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should render checked radio button', () => {
      const wrapper = shallow(
        <RadioButton checked id="radio1" onChange={onChange}>
          Radio 1
        </RadioButton>
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should render radio button', () => {
      const wrapper = shallow(
        <RadioButton id="radio2" onChange={onChange}>
          Radio 2
        </RadioButton>
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
