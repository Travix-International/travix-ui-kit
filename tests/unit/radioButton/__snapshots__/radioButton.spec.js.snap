// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Button #render() should render checked radio button 1`] = `
<div
  className="ui-radio"
>
  <input
    checked={true}
    className="ui-radio__input-radio"
    disabled={false}
    id="radio1"
    onChange={[Function]}
    type="radio"
  />
  <label
    aria-checked={true}
    className="ui-radio__label"
    htmlFor="radio1"
    role="radio"
  >
    Radio 1
  </label>
</div>
`;

exports[`Radio Button #render() should render disabled radio button 1`] = `
<div
  className="ui-radio ui-radio_disabled"
>
  <input
    checked={false}
    className="ui-radio__input-radio"
    disabled={true}
    id="disabledRadio"
    onChange={[Function]}
    type="radio"
  />
  <label
    aria-checked={false}
    className="ui-radio__label"
    htmlFor="disabledRadio"
    role="radio"
  >
    Disabled radio button
  </label>
</div>
`;

exports[`Radio Button #render() should render radio button 1`] = `
<div
  className="ui-radio"
>
  <input
    checked={false}
    className="ui-radio__input-radio"
    disabled={false}
    id="radio2"
    onChange={[Function]}
    type="radio"
  />
  <label
    aria-checked={false}
    className="ui-radio__label"
    htmlFor="radio2"
    role="radio"
  >
    Radio 2
  </label>
</div>
`;
